### PR TITLE
Adding appropriate failure message if image pulling fails

### PR DESCRIPTION
**Pull request by ubhattacharjya**
### Summary

Initially if image pulling failed, it would still print the following message in log 

"Finished pulling container ....". 

This log message was misleading if image pulling failed. So, now we are adding a condition to check if the image pulling failed and then print the appropriate error message.

### Testing

- [x] Builds on Linux (`make release`)
- [x] Builds on Windows (`go build -out amazon-ecs-agent.exe ./agent`)
- [x] Unit tests on Linux (`make test`) pass
- [x] Unit tests on Windows (`go test -timeout=25s ./agent/...`) pass
- [x] Integration tests on Linux (`make run-integ-tests`) pass
- [x] Integration tests on Windows (`.\scripts\run-integ-tests.ps1`) pass
- [x] Functional tests on Linux (`make run-functional-tests`) pass
- [x] Functional tests on Windows (`.\scripts\run-functional-tests.ps1`) pass

New tests cover the changes: 

### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: agent/engine/docker_task_engine.go [hunk 1]
```diff
@@ -731,17 +731,20 @@ func (engine *DockerTaskEngine) concurrentPull(task *apitask.Task, container *ap
...
-	}(pullStart)
 	ok := task.SetPullStartedAt(pullStart)
 	if ok {
 		seelog.Infof("Task engine [%s]: Recording timestamp for starting image pulltime: %s",
 			task.Arn, pullStart)
 	}
-
-	return engine.pullAndUpdateContainerReference(task, container)
+	metadata := engine.pullAndUpdateContainerReference(task, container)
+	if (metadata.Error == nil) {
```
Comment (sharanyad) [inline agent/engine/docker_task_engine.go]: nit: this can be: 
`if metadata.Error == nil {`

File: agent/engine/docker_task_engine.go [hunk 2]
```diff
@@ -731,17 +731,20 @@ func (engine *DockerTaskEngine) concurrentPull(task *apitask.Task, container *ap
...
+		seelog.Infof("Task engine [%s]: Finished pulling container %s in %s",
+			task.Arn, container.Image, time.Since(pullStart).String())
+	} else {
```
Comment (sharanyad) [inline agent/engine/docker_task_engine.go]: is `metadata.Error` a string?

Comment (ubhattacharjya) [reply agent/engine/docker_task_engine.go]: No it is not. It is of type apierrors.NamedError where 
type NamedError interface {
	error
	ErrorName() string
}

File: agent/engine/docker_task_engine.go (hunk 1 above)
Comment (ubhattacharjya) [reply agent/engine/docker_task_engine.go]: Fixed it

File: agent/engine/docker_task_engine.go (hunk 2 above)
Comment (ubhattacharjya) [reply agent/engine/docker_task_engine.go]: No it is not a string. It is of type apierrors.NamedError where
type NamedError interface {
error
ErrorName() string
}

**Pull request merged**